addons/portal/static/src/xml/portal_chatter.xml

```diff
@@ -78,38 +100,52 @@
+                    <hr class="o_portal_chatter_separator_line"/>
+                    <span class="o_portal_chatter_separator_label" t-att-title="msg_group['date']"><t t-esc="msg_group['label']"/></span>
+                    <hr class="o_portal_chatter_separator_line"/>
+                </div>
+                <t t-foreach="msg_group['messages'] || []" t-as="message">
+                    <div class="media o_portal_chatter_message mt-2" t-att-id="'message-' + message.id">
+                        <img class="o_portal_chatter_avatar" t-att-src="message.author_avatar_url" alt="avatar"/>
+                        <div class="media-body">
+                            <div class="o_portal_chatter_message_title row mb-1 mx-0">
+                                <h5 class='mb-0'><t t-esc="message.author_id[1]"/></h5>
+                                <p class="mx-2 pt-1">-</p>
+                                <p class="o_portal_chatter_published_date pt-1" t-att-title="message.published_date_str"><t t-esc="message.date_ago"/></p>
+                                <t t-call="portal.chatter_internal_toggle" t-if="widget.options['is_user_employee']"/>
+                            </div>
+                            <t t-out="message.body"/>
+                            <div t-if="message.attachment_ids.length" class="o_portal_chatter_attachments">
+                                <t t-call="portal.chatter_attachments_link">
+                                    <t t-set="attachments" t-value="message.attachment_ids"/>
+                                    <t t-set="attachment_count" t-value="message.attachment_ids.length"/>
+                                </t>
+                            </div>
                         </div>
                     </div>
-                </div>
+                </t>
             </t>
         </div>
     </t>
 
     <!-- Chatter: internal toggle widget -->
     <t t-name="portal.chatter_internal_toggle">
-        <div t-attf-class="float-right o_portal_chatter_js_is_internal #{message.is_internal and 'o_portal_message_internal_on' or 'o_portal_message_internal_off'}"
+        <div t-attf-class="float-right o_boolean_toggle custom-control custom-checkbox ml-auto #{!message.internal ? 'o_portal_chatter_js_is_internal' : ''}"
                 t-att-data-message-id="message.id"
                 t-att-data-is-internal="message.is_internal">
-            <button class="btn btn-danger"
-                title="Currently restricted to internal employees, click to make it available to everyone viewing this document.">Employees Only</button>
-            <button class="btn btn-success"
-                title="Currently available to everyone viewing this document, click to restrict to internal employees.">Visible</button>
+            <input type="checkbox" t-att-id="'message_' + message.id" class="custom-control-input"
+                t-att-checked="(!message.is_internal and !message.internal and 'checked') or undefined"
```

Comment:
you lost me. internal or not is_internal ? what's the difference ?

Reply:
Seems you are mixing two concepts, subtype being internal ('which cannot be changed) and is_internal (reverse of "publish") ... not sure what is the expected result but seems weird. You will not be able to change the status of messages linked to an internal subtype, as this should be managed at ACL level (never seen by portal users).

Reply:
I am intentionally mixing those concepts.
The boolean_toggle is displayed for an internal user visiting the portal, and it is disabled and unchecked for internal (subtype.internal) messages.
For other messages, it is enabled and uncheckd if is_internal is true, and enabled and checked if is_internal is false.
In other words: checked indicates the user that the message is visible for portal users, unchecked indicates the user that the message is not visible.
disabled indicates the user that the message is an internal log note and will never be visible for portal users.